aten/src/ATen/native/cuda/kernels.cpp

```diff
@@ -0,0 +1,283 @@
+  CHECK_CUSPARSE( cusparseLtMatDescSetAttribute(&handle, &activation_descriptor,
+                                          CUSPARSELT_MAT_NUM_BATCHES,
+                                          &num_batches, sizeof((int)num_batches)) )
+  CHECK_CUSPARSE( cusparseLtMatDescSetAttribute(&handle, &matC,
+                                          CUSPARSELT_MAT_NUM_BATCHES,
+                                          &num_batches, sizeof((int)num_batches)) )
+  //--------------------------------------------------------------------------
+  // SET BATCH STRIDE
+  // if batch_strideA = 0, the matrix multiplication performs a broadcast of
+  // the matrix A
+  CHECK_CUSPARSE(  cusparseLtMatDescSetAttribute(&handle, &weight_descriptor,
+                                              CUSPARSELT_MAT_BATCH_STRIDE,
+                                              &batch_strideA,
+                                              sizeof(batch_strideA)) )
+  CHECK_CUSPARSE(  cusparseLtMatDescSetAttribute(&handle, &activation_descriptor,
+                                              CUSPARSELT_MAT_BATCH_STRIDE,
+                                              &batch_strideB,
+                                              sizeof(batch_strideB)) )
+  CHECK_CUSPARSE(  cusparseLtMatDescSetAttribute(&handle, &matC,
+                                              CUSPARSELT_MAT_BATCH_STRIDE,
+                                              &batch_strideC,
+                                              sizeof(batch_strideC)) )
+  // matmul, algorithm selection, and plan initialization
+  CHECK_CUSPARSE( cusparseLtMatmulDescriptorInit(
+                                          &handle, &matmul, op_weight, op_activation,
+                                          &weight_descriptor, &activation_descriptor, &matC, &matC,
+                                          compute_type) )
+  //--------------------------------------------------------------------------
+  // SET BIAS POINTER
+  auto  dBias = bias.data_ptr<c10::Half>();
+  // CHECK_CUDA( cudaMalloc((void**) &dBias, m * sizeof(float)) )
+  // CHECK_CUDA( cudaMemcpy(dBias, hBias, m * sizeof(float),
+                          // cudaMemcpyHostToDevice) )
+  CHECK_CUSPARSE( cusparseLtMatmulDescSetAttribute(&handle, &matmul,
+                                              CUSPARSELT_MATMUL_BIAS_POINTER,
+                                              &dBias, sizeof(dBias)) )
+
+  int alg = 0;
+  CHECK_CUSPARSE(cusparseLtMatmulAlgSetAttribute(
+      &handle, &alg_sel, CUSPARSELT_MATMUL_ALG_CONFIG_ID, &alg, sizeof(alg)))
+  size_t workspace_size;
+  CHECK_CUSPARSE(
+      cusparseLtMatmulGetWorkspace(&handle, &plan, &workspace_size))
+
+  CHECK_CUSPARSE(cusparseLtMatmulPlanInit(
+      &handle, &plan, &matmul, &alg_sel, workspace_size))
+}
+
+// see https://docs.nvidia.com/cuda/cusparselt/types.html for pruning_algo choices
+void CusparseLtLinear::prune() {
```

Comment:
Can we make this + compress a standalone function that returns an instance of a class such as `CuSparseLtCompressedMatrix`? Then it'll also be easier to understand how to create custom pruning algorithms.

Reply:
Then you can also pass an instance of `CuSparseLtCompressedMatrix` to separate a `cusparselt_addmm(input, cusparselt_weight, bias)` which can immediately be used for F.linear.